File: php/src/Domains/Shortlinks.php
<?php

namespace SignalHouse\SDK\Domains;

use SignalHouse\SDK\HttpClient;

class Shortlinks
{
    private HttpClient $client;
    private bool $enableAdmin;

    public function __construct(HttpClient $client, bool $enableAdmin)
    {
        $this->client = $client;
        $this->enableAdmin = $enableAdmin;
    }

    /**
     * Get the redirect URL for a shortlink by its ID
     *
     * @param string $shortlinkId The shortlink ID
     * @param array $options Additional request options
     * @return array The response from the server
     */
    public function getShortlinkRedirect(string $shortlinkId, array $options = []): array
    {
        $this->client->require(['shortlinkId' => $shortlinkId]);
        $safeShortlinkId = rawurlencode($shortlinkId);
        return $this->client->request("/shortlink/redirect/{$safeShortlinkId}", array_merge([
            'method' => 'GET',
        ], $options));
    }

    /**
     * Get shortlink details with optional filters
     *
     * @param array $params Filter parameters (shortlinkId, messageId, phoneNumber, campaignId, brandId, subgroupId, groupId, page, limit)
     * @param array $options Additional request options
     * @return array The response from the server
     */
    public function getShortlink(array $params = [], array $options = []): array
    {
        $queryString = $this->client->getQueryString($params);
        return $this->client->request("/shortlink{$queryString}", array_merge([
            'method' => 'GET',
        ], $options));
    }
}

File: python/signalhouse/domains/shortlinks.py
"""Shortlinks domain for the SignalHouse SDK."""

from __future__ import annotations

from typing import Any, TYPE_CHECKING
from urllib.parse import quote

if TYPE_CHECKING:
    from ..client import SignalHouseSDK


class Shortlinks:
    """URL shortlink operations."""

    def __init__(self, sdk: SignalHouseSDK) -> None:
        self._sdk = sdk

    def get_shortlink_redirect(
        self,
        shortlink_id: str,
        *,
        token: str | None = None,
        headers: dict[str, str] | None = None,
    ) -> dict[str, Any]:
        """Get the redirect URL for a shortlink by its ID.

        Args:
            shortlink_id: The ID of the shortlink.
            token: Optional bearer token for authentication.
            headers: Additional headers to include in the request.

        Returns:
            Standardized response dict.

        Raises:
            SignalHouseValidationError: If shortlink_id is missing.
        """
        self._sdk._require({"shortlinkId": shortlink_id})
        safe_shortlink_id = quote(str(shortlink_id), safe="")
        return self._sdk._request(
            f"/shortlink/redirect/{safe_shortlink_id}",
            method="GET",
            token=token,
            headers=headers,
        )

    def get_shortlink(
        self,
        *,
        shortlink_id: str | None = None,
        message_id: str | None = None,
        phone_number: str | None = None,
        campaign_id: str | None = None,
        brand_id: str | None = None,
        subgroup_id: str | None = None,
        group_id: str | None = None,
        page: int | None = None,
        limit: int | None = None,
        token: str | None = None,
        headers: dict[str, str] | None = None,
    ) -> dict[str, Any]:
        """Get details of a shortlink with optional filters.

        Args:
            shortlink_id: The ID of the shortlink.
            message_id: Filter by associated message ID.
            phone_number: Filter by associated phone number.
            campaign_id: Filter by associated campaign ID.
            brand_id: Filter by associated brand ID.
            subgroup_id: Filter by associated subgroup ID.
            group_id: Filter by associated group ID.
            page: The page number for pagination.
            limit: The number of items per page.
            token: Optional bearer token for authentication.
            headers: Additional headers to include in the request.

        Returns:
            Standardized response dict.
        """
        query_string = self._sdk._get_query_string({
            "shortlinkId": shortlink_id,
            "messageId": message_id,
            "phoneNumber": phone_number,
            "campaignId": campaign_id,
            "brandId": brand_id,
            "subgroupId": subgroup_id,
            "groupId": group_id,
            "page": page,
            "limit": limit,
        })
        return self._sdk._request(
            f"/shortlink{query_string}",
            method="GET",
            token=token,
            headers=headers,
        )
